### PR TITLE
Huggingface sagemaker test fix

### PR DESCRIPTION
**Pull request by RadhikaB-97**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Temporary change to generate an empty report for huggingface sm local test.
### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/testrunner.py [hunk 1]
```diff
@@ -430,6 +430,12 @@ def main():
             report = os.path.join(os.getcwd(), "test", f"{test_type}.xml")
             sm_utils.generate_empty_report(report, test_type, "neuron")
             return
+        if "huggingface" in dlc_images:
```
Comment (arjkesh) [inline test/testrunner.py:433]: this will skip for all HF - it looks like HF pytorch has local mode tests, same with HF tf inference. We only want to skip HF tf training. "huggingface-tensorflow-training" is likely what you want to skip.

Also please remove the "placeholder" test as well as the whole "local" directory under huggingface_tensorflow, since it is not providing any value and just adding confusion.

File: test/testrunner.py [hunk 2]
```diff
@@ -424,18 +424,14 @@ def main():
...
-            return
-        if "habana" in dlc_images:
-            LOGGER.info(f"Skipping sagemaker tests because Habana is not yet supported on SM. Images: {dlc_images}")
-            # Creating an empty file for because codebuild job fails without it
-            report = os.path.join(os.getcwd(), "test", f"{test_type}.xml")
-            sm_utils.generate_empty_report(report, test_type, "habana")
-            return
+        for image in dlc_images:
+            if image in ["habana","neuron","huggingface-tensorflow-training"]:
+                LOGGER.info(f"Skipping sagemaker tests because Habana is not yet supported on SM. Images: {dlc_images}")
```
Comment (RadhikaB-97) [inline test/testrunner.py:429]: Will change the logger to a general one.

File: test/sagemaker_tests/huggingface_tensorflow/training/integration/local/test_placeholder.py [hunk 3]
```diff
@@ -5,5 +5,6 @@
 @pytest.mark.model("placeholder")
 @pytest.mark.skip_cpu
 @pytest.mark.skip_py2_containers
+#If planning to add any tests remove the huggingface-tensorflow-training from testrunner.py file
```
Comment (arjkesh) [inline test/sagemaker_tests/huggingface_tensorflow/training/integration/local/test_placeholder.py:8]: Nit: Change to 

```
# TODO: Remove huggingface-tensorflow-training skip condition from testrunner before adding tests
```
TODO comment highlights a comment on a lot of text editors

File: test/testrunner.py [hunk 4]
```diff
@@ -424,18 +424,19 @@ def main():
...
-        if "neuron" in dlc_images:
-            LOGGER.info(f"Skipping sagemaker tests because Neuron is not yet supported on SM. Images: {dlc_images}")
+        sm_local_to_skip = {
+            "habana": "Skipping SM tests because SM does not yet support Habana",
+            "neuron": "Skipping - there are no local mode tests for Neuron",
+            "huggingface-tensorflow-training": "Skipping - there are no local mode tests for HF TF training"
+        }
+
+        for skip_condition, reason in sm_local_to_skip.items():
+            LOGGER.info(f"{reason}. Images: {dlc_images}")
```
Comment (arjkesh) [inline test/testrunner.py:434]: There needs to be an if condition here - if skip_condition in dlc_images - right now this is skipping everything

**Review by arjkesh — APPROVED** [review]
